### DISCUSSION TITLE
"KnexTimeoutError: Knex: Timeout acquiring a connection"

### DISCUSSION THREAD

**Discussion by ammaraga** [directus/directus#14059] [Archive]
I keep getting this error.
Directus is running in a K9 pod, DB is an Azure SQL Database.

"KnexTimeoutError: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?"

Directus is mostly working, but some actions are talking very long or I get some error. I suspect this error has something to do with it.

**Comment by ammaraga** [comment]
I believe I found the culprit.
It's always this line that takes ~40 Seconds and slows everything down. Any idea how to deal with it?

request completed GET 200 /fields?limit=-1 41s

**Reply by benhaynes** [reply]
`limit=-1` means "unlimited" (fetch all items, regardless of the count). If you have many items (eg: hundreds of thousands), this will always crash. We would need to know exactly _when_ this happens (your config or our code) to resolve.

**Reply by ammaraga** [reply]
Does this help? The last line is a 39 second wait.

 ││ 19:21:30 ✨ request completed GET 304 /notifications?filter=%7B%22_and%22:[%7B%22recipient%22:%7B%22_eq%22:%22843AC702-7D97-4735-B146-9702BC0BC064%22%7D%7D,%7B%22status%22:%7B% 
 ││ 19:21:30 ✨ request completed GET 200 /server/info 189ms 
 ││ 19:21:30 ✨ request completed GET 304 /collections/test 205ms 
 ││ 19:21:30 ✨ request completed GET 304 /server/info 102ms 
 ││ 19:21:30 ✨ request completed GET 304 /notifications?filter=%7B%22_and%22:[%7B%22recipient%22:%7B%22_eq%22:%22843AC702-7D97-4735-B146-9702BC0BC064%22%7D%7D,%7B%22status%22:%7B% 
 ││ 19:21:30 ✨ request completed PATCH 204 /users/me/track/page 98ms 
 ││ 19:21:31 ✨ request completed GET 304 /admin/assets/Inter-MediumItalic.d4a7f5d9.woff2 
 ││ 19:21:32 ✨ request completed PATCH 204 /users/me/track/page 23ms 
 ││ 19:21:33 ✨ request completed PATCH 204 /users/me/track/page 23ms 
 ││ 19:21:37 ✨ request completed PATCH 204 /users/me/track/page 35ms 
 ││ 19:21:39 ✨ request completed PATCH 204 /users/me/track/page 37ms 
 ││ 19:22:08 ✨ request completed GET 304 /fields?limit=-1 39s
